packages/thrift-server-core/src/main/Int64.ts

```diff
@@ -299,30 +418,29 @@ export class Int64 {
      *
      * @param sep separator string. default is '' (empty string)
      */
-    public toOctetString(sep: string): string {
-        const out = new Array(8)
-        const buf = this.buffer
-        const off = this.offset
-        for (let i = 0; i < 8; i++) {
-            out[i] = _HEX[buf[off + i]]
+    public toOctetString(sep: string = ''): string {
+        if (!sep) {
+            return this.buffer.toString('hex')
         }
-        return out.join(sep || '')
+        return Array.from(this.buffer, (num) => _HEX[num]).join(sep)
     }
 
     /**
      * Returns the int64's 8 bytes in a buffer.
      *
-     * @param {bool} [rawBuffer=false]  If no offset and this is true, return the internal buffer.  Should only be used if
-     *                                  you're discarding the Int64 afterwards, as it breaks encapsulation.
+     * @param {boolean} [rawBuffer=false]  If no offset and this is true, return the internal buffer.
+     *                                     Should only be used if you're discarding the Int64 afterwards,
+     *                                     as it breaks encapsulation.
      */
-    public toBuffer(rawBuffer: Buffer): Buffer {
-        if (rawBuffer && this.offset === 0) {
+    public toBuffer(rawBuffer: boolean = false): Buffer {
```

Comment:
The doc already said this was a boolean, not a Buffer. Looks like somebody may have quickly gone through and added types and guessed the wrong type based on this parameter name. Fixed to match the doc (and the behavior makes sense).